modules/renter.go

```diff
@@ -50,12 +54,92 @@ type Allowance struct {
 // DownloadInfo provides information about a file that has been requested for
 // download.
 type DownloadInfo struct {
-	SiaPath     string    `json:"siapath"`
-	Destination string    `json:"destination"`
-	Filesize    uint64    `json:"filesize"`
-	Received    uint64    `json:"received"`
-	StartTime   time.Time `json:"starttime"`
-	Error       string    `json:"error"`
+	SiaPath     string         `json:"siapath"`
+	Destination DownloadWriter `json:"destination"`
```

Comment:
Unless doing a type switch, caller is not going to be able to see the fields like Location. Given that this struct is only ever used for returning information about a download to the caller, seems strange to me to have this be an interface. Especially because this struct is going to be fed over the API.